Tasks/MavenV3/maventask.ts

```diff
@@ -538,4 +539,9 @@ function processMavenOutput(data) {
     }
 }
 
-execBuild();
+if (restoreOriginalPomXml) {
+    const originalPomContents: string = fs.readFileSync(mavenPOMFile, 'utf8');
+    execBuild().then(() => fs.writeFileSync(mavenPOMFile, originalPomContents));
```

Comment:
Nit: could it make sense to move condition to callback - to avoid duplication?

Reply:
If we move this condition to callback it will look something like this:

```
let originalPomContents: string;
if (restoreOriginalPomXml) {
    originalPomContents = fs.readFileSync(mavenPOMFile, 'utf8');
}

execBuild().then(() => {
    if (restoreOriginalPomXml) {
        fs.writeFileSync(mavenPOMFile, originalPomContents);
    }
});
```

No code duplication, but in my opinion current version looks a bit more expressive:

```
if (restoreOriginalPomXml) {
    const originalPomContents: string = fs.readFileSync(mavenPOMFile, 'utf8');
    execBuild().then(() => fs.writeFileSync(mavenPOMFile, originalPomContents));
} else {
   execBuild();
}
```

If you think that avoiding code duplication is more important here, I'll change this